Sources/Sentry/SentryNetworkTracker.m

```diff
@@ -265,6 +288,121 @@ - (void)urlSessionTask:(NSURLSessionTask *)sessionTask setState:(NSURLSessionTas
+            @"Request url isn't within the request targets, not capturing HTTP Client errors.");
+        return;
+    }
+
+    NSString *message = [NSString
+        stringWithFormat:@"HTTP Client Error with status code: %ld", (long)myResponse.statusCode];
+
+    SentryEvent *event = [[SentryEvent alloc] initWithLevel:kSentryLevelError];
+
+    SentryThreadInspector *threadInspector = SentrySDK.currentHub.getClient.threadInspector;
+    NSArray<SentryThread *> *threads = [threadInspector getCurrentThreads];
+
+    // sessionTask.error isn't used because it's not about network errors but rather
+    // requests that are considered failed depending on the HTTP status code
+    SentryException *sentryException = [[SentryException alloc] initWithValue:message
+                                                                         type:@"HTTPClientError"];
+    sentryException.mechanism = [[SentryMechanism alloc] initWithType:@"HTTPClientError"];
+
+    for (SentryThread *thread in threads) {
+        if ([thread.current boolValue]) {
+            SentryStacktrace *sentryStacktrace = [thread stacktrace];
+            sentryStacktrace.snapshot = @(YES);
+
+            sentryException.stacktrace = sentryStacktrace;
+
+            break;
+        }
+    }
+
+    SentryRequest *request = [[SentryRequest alloc] init];
+
+    NSURL *url = [[sessionTask currentRequest] URL];
+
+    NSString *urlString = [NSString stringWithFormat:@"%@://%@%@", url.scheme, url.host, url.path];
+
+    request.url = urlString;
+    request.method = myRequest.HTTPMethod;
+    request.fragment = url.fragment;
+    request.queryString = url.query;
+    if (sessionTask.countOfBytesSent != 0) {
+        request.bodySize = [NSNumber numberWithLongLong:sessionTask.countOfBytesSent];
+    }
+    if (nil != myRequest.allHTTPHeaderFields) {
+        NSDictionary<NSString *, NSString *> *headers = myRequest.allHTTPHeaderFields.copy;
+        request.headers = headers;
+        request.cookies = headers[@"Cookie"];
+    }
+
+    event.exceptions = @[ sentryException ];
+    event.request = request;
+
+    NSMutableDictionary<NSString *, id> *context = [[NSMutableDictionary alloc] init];
+    NSMutableDictionary<NSString *, id> *response = [[NSMutableDictionary alloc] init];
+
+    [response setValue:responseStatusCode forKey:@"status_code"];
+    if (nil != myResponse.allHeaderFields) {
+        NSDictionary<NSString *, NSString *> *headers = myResponse.allHeaderFields.copy;
+        [response setValue:headers forKey:@"headers"];
+        [response setValue:headers[@"Cookie"] forKey:@"cookies"];
+    }
+    if (sessionTask.countOfBytesReceived != 0) {
+        [response setValue:[NSNumber numberWithLongLong:sessionTask.countOfBytesReceived]
+                    forKey:@"body_size"];
+    }
+
+    context[@"response"] = response;
+    event.context = context;
+
+    // TODO: do we need to pass the SentrySDK.currentHub.scope here?
+    [SentrySDK captureEvent:event];
```

Comment:
No, I don't think so. Why would you want to pass the current scope? `SentrySDK captureEvent` already uses the current scope.

Reply:
I dont want to, but I saw other parts of the code doing it.